pymeasure/instruments/instrument.py

```diff
@@ -33,6 +33,60 @@
 log = logging.getLogger(__name__)
 log.addHandler(logging.NullHandler())
 
+reserved_prefix = "___"  # Prefix used to store reserved variables
+
+
+class DynamicProperty(property):
+    """ Class that allows to manage python property behaviour in a "dynamic" fashion
+
+    The class allows to pass, in addition to regular property parameters, a list of
+    runtime configurable parameters.
+    The effect is that the behaviour of fget/fset not only depends on the obj parameter, but
+    also on a set of keyword parameters with a default value.
+    These extra parameters are read from instance, if available, or left with the default value.
+    Dynamic behaviour is achieved by changing class or instance variables with special names.
+
+    :param fget_list: List of parameter's names that are dynamically configurable
+    :param fset_list: List of parameter's names that are dynamically configurable
+    :param fget: class property fget parameter with added parameters as in fget_list
+    :param fset: class property fget parameter with added parameters as in fget_list
+    :param fdel: class property fdel parameter
+    :param doc: class property doc parameter
+    """
```

Comment:
Am I guessing correctly that https://docs.python.org/3/howto/descriptor.html#properties was the template for this subclass implementation? Might be useful in any case to put a link to that in here for reference.

Reply:
Indeed, it was a while ago, but your comment made me remember this. I will put the reference.

Reply:
Done